src/main/java/com/saralein/server/connection/ConnectionHandler.java

```diff
@@ -1,29 +1,39 @@
 package com.saralein.server.connection;
 
 import com.saralein.server.logger.Logger;
-import com.saralein.server.response.Response;
+import com.saralein.server.request.Request;
+import com.saralein.server.request.RequestParser;
+import com.saralein.server.router.Router;
 
 import java.io.IOException;
+import java.util.HashMap;
 
 public class ConnectionHandler implements Runnable {
     private final Logger logger;
     private final Connection socket;
-    private final Response responseBuilder;
+    private final Router router;
+    private final RequestParser requestParser;
 
-    public ConnectionHandler(Connection socket, Logger logger, Response responseBuilder) {
+    public ConnectionHandler(Connection socket, Logger logger, Router router, RequestParser requestParser) {
         this.socket = socket;
         this.logger = logger;
-        this.responseBuilder = responseBuilder;
+        this.router = router;
+        this.requestParser = requestParser;
     }
 
-    public void sendResponse(Connection socket) throws IOException {
-        byte[] response = responseBuilder.createResponse();
+    private HashMap<String, String> getRequest(Connection socket) throws IOException {
+        return requestParser.parse(socket.read());
+    }
+
+    private void sendResponse(Connection socket, byte[] response) throws IOException {
         socket.write(response);
     }
```

Comment:
this function doesn't seem necessary anymore since it's just one line, basically a static function, and all the members present inside the body are just passed to it

Reply:
[Removed that function](https://github.com/saralein/java-server/pull/3/commits/c4ece7364ce710f6e9d79c924e8524668a9cd438#diff-982b340a061d14794abbf3b365dcc238R27).  What about `getRequest`?  I suppose that also seems a little silly to me now.